src/instructlab/system/sysinfo.py

```diff
@@ -0,0 +1,146 @@
+        "torch.cuda.current": torch.cuda.current_device(),
+    }
+
+    for idx in range(torch.cuda.device_count()):
+        device = torch.device("cuda", idx)
+        free, total = torch.cuda.mem_get_info(device)
+        capmax, capmin = torch.cuda.get_device_capability(device)
+        info[f"torch.cuda.{idx}.name"] = torch.cuda.get_device_name(device)
+        info[f"torch.cuda.{idx}.free"] = f"{(free / 1024**3):.1f}"
+        info[f"torch.cuda.{idx}.total"] = f"{(total / 1024**3):.1f}"
+        info[f"torch.cuda.{idx}.capability"] = f"{capmax}.{capmin}"
+
+    return info
+
+
+def _torch_hpu_info() -> typing.Dict[str, typing.Any]:
+    """Intel Gaudi (HPU) devices"""
+    # Third Party
+    import torch
+
+    try:
+        # Third Party
+        from habana_frameworks.torch import hpu
+    except ImportError:
+        return {}
+
+    info = {
+        # 'habana_frameworks' has package name 'habana_torch_plugin'
+        "habana_torch_plugin.version": importlib.metadata.version(
+            "habana_torch_plugin"
+        ),
+        "torch.hpu.is_available": hpu.is_available(),
+    }
+
+    if not info["torch.hpu.is_available"]:
+        return info
+
+    info["torch.hpu.device_count"] = hpu.device_count()
+    for idx in range(hpu.device_count()):
+        device = torch.device("hpu", idx)
+        info[f"torch.hpu.{idx}.name"] = hpu.get_device_name(device)
+        info[f"torch.hpu.{idx}.capability"] = hpu.get_device_capability(device)
+        prop: str = hpu.get_device_properties(device)
+        info[f"torch.hpu.{idx}.properties"] = prop.strip("()")
+    for key, value in sorted(os.environ.items()):
+        if key.startswith(("PT_", "HABANA", "LOG_LEVEL_", "ENABLE_CONSOLE")):
+            info[f"env.{key}"] = value
+    return info
+
+
+def _llama_cpp_info() -> typing.Dict[str, typing.Any]:
+    """llama-cpp-python capabilities"""
+    # Third Party
+    import llama_cpp
+
+    return {
+        "llama_cpp_python.version": importlib.metadata.version("llama_cpp_python"),
+        "llama_cpp_python.supports_gpu_offload": llama_cpp.llama_supports_gpu_offload(),
+    }
+
+
+def _instructlab_info():
+    """InstructLab packages"""
+    # auto-detect all instructlab packages
+    pkgs = sorted(
+        (dist.name, dist.version)
+        for dist in importlib.metadata.distributions()
+        if dist.name.startswith("instructlab")
+    )
+    return {f"{name}.version": ver for name, ver in pkgs}
+
+
+def get_sysinfo() -> typing.Dict[str, typing.Any]:
+    """Get system information"""
+    info = {}
+    info.update(_platform_info())
+    info.update(_instructlab_info())
+    info.update(_torch_info())
+    info.update(_torch_cuda_info())
+    info.update(_torch_hpu_info())
+    info.update(_llama_cpp_info())
+    return info
+
+
+@click.command()
+def sysinfo():
+    """Print system information"""
+    for key, value in get_sysinfo().items():
+        print(f"{key}: {value}")
```

Comment:
perhaps consider click.echo?

Reply:
non-blocking

Reply:
Sure I'll switch this to use `click.echo`. I just copied this over from the previous file, so this isn't my code.

Reply:
`click.echo` as a replacement for `print` is a bit overkill. I can refactor the code in another PR to use `click.echo` with additional bells and whistles like colors to highlight the key.

Reply:
Either way is fine with me. Marking as resolved for now as it sounds like this is a separate PR problem.